src/libawkward/Content.cpp

```diff
@@ -891,6 +891,56 @@ namespace awkward {
     return form(true).get()->branch_depth();
   }
 
+  const ContentPtr
+  Content::reverse_merge(const ContentPtr& other) const {
+    throw std::runtime_error(
+      std::string("undefined operation: ") + classname() + ("::reverse_merge")
+      + FILENAME(__LINE__));
+  }
+
+  const std::pair<ContentPtrVec, ContentPtrVec>
```

Comment:
@jpivarski - I often need to look it up what is `first` and what is `second` in a pair :-) 

 I wonder if using a `struct` here would be more descriptive. For example:
```c++
struct ContentMerger {
  ContentPtrVec head;
  ContentPtrVec tail;
}
```
line 940 in `src/libawkward/array/ListArray.cpp`  would become:
```c++
auto head_tail = merging_strategy(others);
ContentPtrVec head = head_tail.head;
ContentPtrVec tail = head_tail.tail;
```

Reply:
Thanks for the suggestion! In this case, however, we're just splitting a sequence into two subsequences while maintaining order; the first is the first elements and the last is the last elements. Also, it's an internal intermediary, not a user interface (where I worry about these things more).

There are cases like `branch_depth` that may be more difficult to remember because there is no intrinsic ordering to "does this array branch? (bool)?" and "what is its depth, if it doesn't? (int)". However, the fact that they're different types lets the compiler check misinterpretations.